server/publications/collections/product-publications.app-test.js

```diff
@@ -88,14 +108,35 @@ describe("Publication", function () {
         sandbox.stub(Reaction, "getShopId", () => shopId);
         sandbox.stub(Roles, "userIsInRole", () => true);
         sandbox.stub(Reaction, "hasPermission", () => true);
-        sandbox.stub(Reaction, "getShopsWithRoles", () => [shopId]);
+        sandbox.stub(Reaction, "getShopsWithRoles", () => [shopId, primaryShopId]);
 
-        const collector = new PublicationCollector({ userId: Random.id() });
         let isDone = false;
 
         collector.collect("Products", 24, undefined, {}, (collections) => {
           const products = collections.Products;
-          expect(products.length).to.equal(3);
+
+          expect(products.length).to.equal(4);
+
+          if (!isDone) {
+            isDone = true;
+            done();
+          }
+        });
+      });
+
+      it("returns products from only the shops for which an admin has createProduct Role", function (done) {
```

Comment:
the diff is a little strange here, but this is a new test.
the user does not have the createProduct role in the "current" shop but does for other.
It is a bit of a contrived example, but I will make it more realistic in the next PR addressing the #4092